src/AutoRest.CSharp/Mgmt/Decorator/BodyParameterNormalizer.cs

```diff
@@ -45,56 +46,55 @@ internal static void UpdateUsingReplacement(RequestParameter bodyParameter, IDic
                 schemaName = schemaName.ReplaceLast("Info", Content);
             if (schemaName.EndsWith("Input", StringComparison.Ordinal))
                 schemaName = schemaName.ReplaceLast("Input", Content);
-            var paramName = NormalizeParamNames.GetNewName(bodyParameter.Language.Default.Name, schemaName, resourceDataDictionary);
+            var paramName = NormalizeParamNames.GetNewName(bodyParameter.Name, schemaName, resourceDataDictionary);
             // TODO -- we need to add a check here to see if this rename introduces parameter name collisions
-            UpdateRequestParameter(bodyParameter, paramName, schemaName, operation);
+            UpdateRequestParameter(bodyParameter, paramName, schemaName, operation, updatedTypes);
         }
 
-        internal static void UpdateParameterNameOnly(RequestParameter bodyParam, IDictionary<string, HashSet<OperationSet>> resourceDataDictionary, Operation operation)
+        internal static void UpdateParameterNameOnly(InputParameter bodyParam, IDictionary<string, HashSet<OperationSet>> resourceDataDictionary, InputOperation operation)
         {
-            string oriName = bodyParam.Language.Default.Name;
-            bodyParam.Language.Default.SerializedName ??= bodyParam.Language.Default.Name;
-            bodyParam.Language.Default.Name = NormalizeParamNames.GetNewName(bodyParam.Language.Default.Name, bodyParam.Schema.Name, resourceDataDictionary);
+            string oriName = bodyParam.Name;
+            bodyParam.Name = NormalizeParamNames.GetNewName(bodyParam.Name, bodyParam.Type.Name, resourceDataDictionary);
```

Comment:
`bodyParam.Type.Name` is causing parameter name changing from `anyObject` to `unknown`.
Because we are changing from `AnyObjectSchame` to `InputIntrinsicType.Unknown`.